.github/workflows/close-old-pr.yml

```diff
@@ -1,36 +1,70 @@
-name: Close Old Pull Requests
+name: Close Old Issues
 
 on:
   schedule:
-    - cron: '0 0 * * *'  
+    - cron: '0 0 * * *'  # Run daily at midnight UTC
 
 permissions:
-  pull-requests: write  
+  issues: write
+
 jobs:
-  close-old-prs:
+  close-old-issues:
     runs-on: ubuntu-latest
     steps:
-      - name: Close Pull Requests Older Than 30 Days
-        uses: actions/github-script@v6  
+      - name: Close Issues Older Than 30 Days
+        uses: actions/github-script@v6
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
           script: |
-            const daysOld = 30; // Customize the number of days to consider "old"
+            const daysOld = 30; // Number of days to consider "old"
+            const excludeLabels = ['pinned', 'do-not-close']; // Labels to exclude
             const daysAgo = new Date(Date.now() - (1000 * 60 * 60 * 24 * daysOld));
+            const owner = context.repo.owner;
+            const repo = context.repo.repo;
 
-            const { data: prs } = await github.pulls.list({
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              state: 'open',
-              since: daysAgo.toISOString(),
-            });
-
-            for (const pr of prs) {
-              console.log(`Closing pull request #${pr.number} (last updated: ${pr.updated_at})`);
-              await github.pulls.update({
-                owner: context.repo.owner,
-                repo: context.repo.repo,
-                pull_number: pr.number,
-                state: 'closed',
+            try {
+              // Fetch open issues updated before the threshold
+              const { data: issues } = await github.issues.listForRepo({
+                owner,
+                repo,
+                state: 'open',
+                per_page: 100,  // Max per page
+                since: daysAgo.toISOString(),
               });
+
+              // Filter issues that are older than 30 days
+              const oldIssues = issues.filter(issue => 
```

Comment:
add this logic when we first retrieve all the issues. add this in filter

Reply:
Hi @Sibam-Paul @kartikayasijaa ,  Thanks for being patient.
I have added the retrieve logic when retrieving all the issues as per you request

![image](https://github.com/user-attachments/assets/48d846d8-9f60-4d91-bb15-b1cc610105bc)

Reply:
ok